numpy/core/src/common/npy_cpu_features.c

```diff
@@ -43,9 +45,30 @@ npy_cpu_init(void)
     if (npy__cpu_validate_baseline() < 0) {
         return -1;
     }
-    if (npy__cpu_try_disable_env() < 0) {
+    uint8_t env_flags = 0;
+    char *enable_env = getenv("NPY_ENABLE_CPU_FEATURES");
+    if (enable_env != NULL && enable_env[0] != 0) {
+        env_flags += 1;
+        env_flags <<= 1;
+    }
+    char *disable_env = getenv("NPY_DISABLE_CPU_FEATURES");
+    if (disable_env != NULL && disable_env[0] != 0) {
+        env_flags += 1;
+    }
+    if (env_flags == 3){
+        PyErr_Format(PyExc_ImportError,
+            "Both NPY_DISABLE_CPU_FEATURES and NPY_ENABLE_CPU_FEATURES "
+            "environment variables cannot be set simultaneously."
+        );
         return -1;
     }
+    if (env_flags != 3 && env_flags) {
+        int disable = env_flags & 1;
+        char* env =  disable ? disable_env : enable_env;
+        if (npy__cpu_check_env(disable, env) < 0) {
+            return -1;
+        }
+    }    
```

Comment:
```suggestion
    char *enable_env = getenv("NPY_ENABLE_CPU_FEATURES");
    char *disable_env = getenv("NPY_DISABLE_CPU_FEATURES");
    int is_enable = enable_env && enable_env[0];
    int is_disable = disable_env && disable_env[0];
    if (is_enable & is_disable) {
        PyErr_Format(PyExc_ImportError,
            "Both NPY_DISABLE_CPU_FEATURES and NPY_ENABLE_CPU_FEATURES "
            "environment variables cannot be set simultaneously."
        );
        return -1;
    }
    if (is_enable | is_disable) {
        if (npy__cpu_check_env(is_disable, is_disable ? disable_env : enable_env) < 0) {
            return -1;
        }
    }
```
keep it simple no need for nested logical var, shifting, or increments.